File: 2025/647. Palindromic Substrings/Solution.java
class Solution {
    public int countSubstrings(String s) {
        int res = 0;
        for (int i = 0; i < s.length(); i++) {
            res += countPalindrome(s, i, i);
            res += countPalindrome(s, i, i + 1);
        }
        return res;
    }

    private int countPalindrome(String s, int l, int r) {
        int count = 0;
        while (l >= 0 && r < s.length()) {
            if (s.charAt(l--) != s.charAt(r++)) {
                return count;
            }
            count++;
        }
        return count;
    }
}

File: 2025/647. Palindromic Substrings/solution.py
class Solution:
    def countSubstrings(self, s: str) -> int:
        total = 0
        def buildPalindrome(s: str, start: int, end: int) -> int:
            count = 0
            while start >= 0 and end < len(s) and s[start] == s[end]:
                count += 1
                start -= 1
                end += 1
            return count

        for i in range(len(s)):
            total += buildPalindrome(s, i, i)
            if i > 0 and s[i] == s[i - 1]:
                total += buildPalindrome(s, i - 1, i)
        
        return total
